oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -57,13 +59,16 @@
  *
  * <p>Fetches access tokens from the Google Compute Engine metadata server.
  */
-public class ComputeEngineCredentials extends GoogleCredentials {
+public class ComputeEngineCredentials extends GoogleCredentials implements ServiceAccountSigner {
```

Comment:
I need to refresh my memory of the class hierarchy before giving a final opinion, but I'm concerned about adding new interfaces to classes as opposed to making functionality more compositional.

Reply:
The same approach was used in the `AppEngineCreadentials`: https://github.com/google/google-auth-library-java/blob/51a5445b33d10f252cadfdcca82dd9e68512e483/appengine/java/com/google/auth/appengine/AppEngineCredentials.java#L55

Reply:
Ok this seems fine then!